facturx/facturx.py

```diff
@@ -147,3 +146,24 @@ def write_xml(self, path):
         with open(path, 'wb') as f:
             f.write(self.xml_str)
 
+    def __make_dict(self):
+        fields_data = xml_flavor.FIELDS
+        flavor = self.flavor.name
+
+        output_dict = {}
+        for field in fields_data.keys():
+            try:
+                r = self.xml.xpath(fields_data[field]['_path'][flavor], namespaces=self._namespaces)
+                output_dict[field] = r[0].text
+            except IndexError:
+                output_dict[field] = None
+
+        return output_dict
+
+    def write_json(self, json_file_path='output.json'):
+        json_output = self.__make_dict()
+        if json_output is None:
```

Comment:
How can the return value of `__make_dict()` ever be empty? In the worst case it returns an empty dict, but never ever `None`. Maybe this is leftover code from your previous version?